### PR TITLE
[ENH] s-SNOM FFT Processing: Class complex fft part2

### PR DESCRIPTION
**Pull request by levandoskije**
Hi @stuart-cls !
I did some testing and it is possible to use the standard apodization and centering available.
So, I modified the code and now we have only one function to do this pre-processing.
This PR is a continuation of  PR #440  , issue #416  and I fixed some minor errors from the old one.

 - Blocked spacing editing;
 - Blocked sweep editing
 - Apodization;
 - Centralization;
 - Zero filling;
 - Update tests;

### CODE REVIEW

**Comment by stuart-cls** [comment]
Great, thanks @levandoskije ! I will try to look at this soon.

File: orangecontrib/spectroscopy/irfft.py [hunk 1]
```diff
@@ -373,22 +347,18 @@ def __call__(self, ifg, zpd=None, phase=None):
         Nzff = self.zff*number_of_points
```
Comment (stuart-cls) [inline orangecontrib/spectroscopy/irfft.py:347]: This is no longer strictly true, as the zero_fill() function will pad to the nearest power of 2 for DFT efficiency.
I suggest putting Nzff = ifg.shape[0] after the zero_fill() on L357.
Of course check this for your particular calculations.

**Review by stuart-cls — COMMENTED** [review]
Hi @levandoskije ,
Please accept my apologies for the extremely tardy review.

This looks great! I only have one comment about the final zero-filling size used to calculate the wavenumbers. Also you can drop the scipy import now.

I'm curious why the test values changed? Is it just small differences in the calculated window?

In general, I think at this point we could recombine the two classes in a nice way but let's not let that hold up merging this :)

**Comment by levandoskije** [comment]
> Hi @levandoskije ,
> Please accept my apologies for the extremely tardy review.
> 
> This looks great! I only have one comment about the final zero-filling size used to calculate the wavenumbers. Also you can drop the scipy import now.
> 
> I'm curious why the test values changed? Is it just small differences in the calculated window?
> 
> In general, I think at this point we could recombine the two classes in a nice way but let's not let that hold up merging this :)

Hi @stuart-cls ,
The tests change because in the old code when I centralize the spectrum, zeros are added on the side and cut off the other, this can introduce peaks in the final result. Called the "Peak Fence Effect", but with your code, it doesn't happen, your code is more efficient than mine hahahaha

**Comment by stuart-cls** [comment]
Ha gotcha, makes sense. Thanks for the update, I'll push a fake PR to trigger the CI tests and if all looks good I'll merge.

Thanks for your hard work!

**Pull request merged**